grype/pkg/package.go

```diff
@@ -204,6 +204,8 @@ func dataFromPkg(p syftPkg.Package) (interface{}, []UpstreamPackage) {
 		metadata = golangMetadataFromPkg(p)
 	case syftPkg.DpkgDBEntry:
 		upstreams = dpkgDataFromPkg(p)
+	case syftPkg.DpkgArchiveEntry:
```

Comment:
I assume these changes are intentional, and were just missed as part of an early dpkg cataloger PR?

Reply:
these are intentional changes and are needed for tests to pass. Since syft was updated in this PR it included the dpkg work, and the completion-tester automatically detects when there are new metadata types that don't have test cases and automatically fails.